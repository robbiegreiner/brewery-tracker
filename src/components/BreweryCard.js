import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import beer1 from '../assets/tap.svg';

const BreweryCard = ({ brewery, getBrewery, addFavoriteBrewery, user, removeFavoriteBrewery, favorites }) => {
  if(user.email) {
    return (
      <div className={ brewery.isFav ? 'beer-card brewery-card favorite-card' : 'beer-card brewery-card' }>
        <Link to={'/brewery/' + brewery.id}>
          <h2 onClick={() => getBrewery(brewery.id)}>{brewery.name}</h2>
        </Link>
        <h3>{brewery.locations ? brewery.locations[0].locality + ", " + brewery.locations[0].region : ''}</h3>
        {brewery.images ? <img className='brewery-logo' src={brewery.images.squareMedium}></img>: <img className='brewery-icon' src={beer1}></img>}
        <a className='website' href={brewery.website}>{brewery.website}</a>
        <h3 className='type'>{brewery.locations ? brewery.locations[0].locationTypeDisplay : ''}</h3>
        { removeFavoriteBrewery ? <h4 className='favorite' onClick={() => removeFavoriteBrewery(user.id, brewery.firebaseID)}>UNFAVORITE</h4> : <h4 className='favorite' onClick={() => addFavoriteBrewery(favorites, user.id, 'brewery', brewery.id, brewery)}>FAVORITE</h4> }
      </div>
    );
  } else {
    return (
      <div className={ brewery.isFav ? 'beer-card brewery-card favorite-card' : 'beer-card brewery-card' }>
        <Link to={'/brewery/' + brewery.id}>
          <h2 onClick={() => getBrewery(brewery.id)}>{brewery.name}</h2>
        </Link>
        <h3>{brewery.locations ? brewery.locations[0].locality + ", " + brewery.locations[0].region : ''}</h3>
        {brewery.images ? <img className='brewery-logo' src={brewery.images.squareMedium}></img>: <img className='brewery-icon' src={beer1}></img>}
        <a className='website' href={brewery.website}>{brewery.website}</a>
        <h3 className='type'>{brewery.locations ? brewery.locations[0].locationTypeDisplay : ''}</h3>
        <Link to='/login'><h4 className='favorite'>FAVORITE</h4></Link> 
      </div>
    );
  }
};

BreweryCard.propTypes = {
  brewery: PropTypes.object,
  getBrewery: PropTypes.func,
  writeUserData: PropTypes.func,
  user: PropTypes.object,
  removeFavorite: PropTypes.func,
  removeFavoriteBrewery: PropTypes.func
};

export default BreweryCard;
