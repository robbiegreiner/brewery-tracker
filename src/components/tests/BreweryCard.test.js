import React from "react";
import BreweryCard from '../BreweryCard';
import { shallow } from "enzyme";

describe('BreweryCard component', () => {
  const mockFn = jest.fn();

  const wrapper = shallow(<BreweryCard
    user={{}}
    writeUserData={mockFn}
    getBrewery={mockFn}
    brewery={{}}
    key={'vxySde'}/>);


  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


});
