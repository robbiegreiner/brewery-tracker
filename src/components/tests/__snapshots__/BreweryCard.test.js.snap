// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreweryCard component should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BreweryCard
    brewery={Object {}}
    getBrewery={[Function]}
    user={Object {}}
    writeUserData={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          replace={false}
          to="/brewery/undefined"
>
          <h2
                    onClick={[Function]}
          />
</Link>,
        <h3 />,
        <h4
          onClick={[Function]}
>
          FAVORITE
</h4>,
      ],
      "className": "beer-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <h2
            onClick={[Function]}
/>,
          "replace": false,
          "to": "/brewery/undefined",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "FAVORITE",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "FAVORITE",
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            replace={false}
            to="/brewery/undefined"
>
            <h2
                        onClick={[Function]}
            />
</Link>,
          <h3 />,
          <h4
            onClick={[Function]}
>
            FAVORITE
</h4>,
        ],
        "className": "beer-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <h2
              onClick={[Function]}
/>,
            "replace": false,
            "to": "/brewery/undefined",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "FAVORITE",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "FAVORITE",
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
